test/integration-tests/IntegrationTests.Helpers/ProcessHelper.cs

```diff
@@ -0,0 +1,59 @@
+using System;
+using System.Diagnostics;
+using System.Text;
+using System.Threading;
+
+namespace IntegrationTests.Helpers
+{
+    /// <summary>
+    /// Drains the standard and error output of a process
+    /// </summary>
+    public class ProcessHelper : IDisposable
+    {
+        private readonly ManualResetEventSlim _errorMutex = new();
+        private readonly ManualResetEventSlim _outputMutex = new();
+        private readonly StringBuilder _outputBuffer = new();
+        private readonly StringBuilder _errorBuffer = new();
+
+        public ProcessHelper(Process process)
+        {
+            process.OutputDataReceived += (_, e) => DrainOutput(e, _outputBuffer, _outputMutex);
+            process.ErrorDataReceived += (_, e) => DrainOutput(e, _errorBuffer, _errorMutex);
+
+            process.BeginOutputReadLine();
+            process.BeginErrorReadLine();
+        }
+
+        public string StandardOutput => _outputBuffer.ToString();
+
+        public string ErrorOutput => _errorBuffer.ToString();
+
+        public bool Drain(int timeout = Timeout.Infinite)
+        {
+            if (timeout != Timeout.Infinite)
+            {
+                timeout /= 2;
+            }
+
+            return _outputMutex.Wait(timeout) && _errorMutex.Wait(timeout);
+        }
+
+        public void Dispose()
+        {
+            _errorMutex.Dispose();
+            _outputMutex.Dispose();
+        }
+
+        private static void DrainOutput(DataReceivedEventArgs e, StringBuilder buffer, ManualResetEventSlim mutex)
+        {
+            if (e.Data == null)
+            {
+                mutex.Set();
```

Comment:
I like golang style instead of else branch. It can help with size and branch prediction. Not that important on test code but gets us in the good habits. 
```c#
            var data = e.Data;
            if (data != null)
            {
                buffer.AppendLine(data);
                return;
            }
            
            mutex.Set();
```
See [IL/JITAsm](https://sharplab.io/#v2:EYLgtghglgdgNAFxBAzmOATEBqAPgAQCYBGAWACh8AGAAn2IDoARKCAcxgHsUEoBjFAG4K1OowAqAUwAeCYZVr0G4gBYAnSRAyw28ijAhhJKAA4Q+kmgEkYCSWzURenGFJ4oGACUkAbE5LUUCgBvChpwmgB6aJoAHhQAVzBINQBPAD4wiOjImiZHWBQaBBVLHggYDAg1DBoK2oC1TjUaTgSEE3bWgDM6mhMmixQg8giomNjIxOTqjKzw/ABmOkIaAAVB4xRvPwCaEGsWU24IYB9JeZpQ0bGIgagANydLDS0XH1SaAFkKhIgfABKxkkCAAog9JLYAMo+KBgGgAfUazS+7RkNAAvDQYJIAO4ACgAlPJbnc1I9njRXhh3p8fjA/oDgWCIdDYfCEW0Ou1UXZpJjsXiiSTSf1yU87FTNDSYB8aFCEOSYGwAEIJKA+DB7TntToINXdbp7LE4gnEy5je4Sl7S2nyxU6NUarUtJFqJpqA1GlomoXm8gWiJLdabYY7fxqfEbThDIoDGNbQmB8LXUW3eOxhgAeV17SYTggQIsj0ktWwWPxCLgNEkhMx6TyBRgOe5CHxkmrOtbXoCna5et5Mn9actoY8oPdzXzCELkmLELLFarNbrGIb+WgzdzbY7iORnoSht7e8nakH0n9yfTY4YKvssBbeqBWgAMrBJMKr6OE8Nb/eYBOHrPhgb44p+NykgAvhQV7BvQtAKvU1QYI+XRroi/btD2ajKJwCpKmw4Gwcs8E0IBzSoQg9Ynh62G4fhOhERBtzBsAnCcD4jabvisBUbwRhcgK4hwpIXIMDY3SwFAdhJsxYypiO4RQL0+L8aJXQAIRYsJAntOJMCSTA0m1l+EQKYpYxqYJkRYoQIqKdBAZyaK+AAOwYdu54MAA6tAbZWe0dYAGRBTRKJotIPl+apIlcsOoqOcRdAACx5FAxwoB+slpuZI5uh6XlHCY3BZfZeWYQghXpcVmXgWmiXOeEVqUvQABsKVcQ+274tOs7zqW4KQggACCahsEUu4McqTqanswCHt61b0oyQKZSyQ0wnCNBgBF2WirlabKTQ7bMAWmImgkPg+HtikHRZO18gwUIgnVFmORZNY+Jlpkpj9tzzUeOHDSY/iVKBH6SKdM7xQ5V4NWmzWSm1HUAOIcRUbAKqk5wbl1rY9QWRaSCWGCDbYo3jTW1ZTaq6qzS0AOLd8vz/KtIJkwgm3wg9Q5XndpJPC0VQzgKkO9WVh0qcLEA0Fp2KXddf38yOjMBAwwOgyB774tLMMfXQrkSwlf1/TzkXPW2etjO9ESOZBQA===)

Reply:
In the case above the IL is a tiny bit smaller but the ASM is the same size :) bad example.

Reply:
Also more readable.... but there were bigger changes anyway.